restapi/provider.go

```diff
@@ -179,6 +210,19 @@ func configureProvider(d *schema.ResourceData) (interface{}, error) {
 	if v, ok := d.GetOk("destroy_method"); ok {
 		opt.destroy_method = v.(string)
 	}
+	if _, ok := d.GetOk("oauth_client_credentials"); ok {
+		opt.use_oauth_client_credentials = true
+	}
+	if v, ok := d.GetOk("oauth_client_credentials"); ok {
+		opt.use_oauth_client_credentials = true
```

Comment:
I'd propose removing this boolean and use idiomatic go methods to determine if this should be used.

For example, since these are all required attributes of `oauth_client_credentials`, we know they'll be set and could blindly pass them along as `opt` items.

Reply:
👍